docs/tutorial/project_configuration.md

```diff
@@ -0,0 +1,56 @@
+Use the `ESP-IDF: Open Project Configuration` to manage the project configuration profiles to record the following settings for each configuration:
+
+`idf.cmakeCompilerArgs`
+`idf.ninjaArgs`
+`idf.buildPath`
+`idf.sdkconfigFilePath`
+`idf.sdkconfigDefaults`
+
+`idf.customExtraVars`
+`idf.adapterTargetName`
+`idf.flashBaudRate`
+`idf.monitorBaudRate`
+
+`idf.openOcdDebugLevel`
+`idf.openOcdConfigs`
+`idf.openOcdLaunchArgs`
+
+`idf.preBuildTask`
+`idf.postBuildTask`
+`idf.preFlashTask`
+`idf.postFlashTask`
+
+After defining a profile and the settings for each profile use the `ESP-IDF: Select project configuration` command to choose the configuration to override extension configuration settings.
+
+There are many use cases for having multiple configurations profiles. It allows the user to store settings together and easily switch between one and the other. Let's explore one of this use cases here, having a development and production build profiles.
+
+# Development and Release profiles for ESP-IDF project
+
+In the ESP-IDF CMake build system, the project configuration settings are saved using the `idf.py menuconfig` (or the SDK Configuration editor in IDEs) which store these values in a `sdkconfig` file. This file is used by the build system and the source code. When the current ESP-IDF project is under version control system, the `sdkconfig` can change on any user build due to many local environment configuration which can alter the project expected behavior. For such a reason is better to move those project specific settings to an `sdkconfig.defaults` which is not modified by the build system and `sdkconfig` can be added to the `.gitignore` list. This `sdkconfig.defaults` can be generated by `idf.py save-defconfig`.
+
+The default case, without using any project configuration editor setting would be to create an `sdkconfig` file in the ESP-IDF project root directory. Initially the user would use the default case but later it would like to commit the project for version control. Since `sdkconfig` contains many values that are not specific to the project, the user would like to save only those values that are required for the project. These values would be the first **production** profile.
+
+1. For the current project, run the `ESP-IDF: Open ESP-IDF Terminal` command to launch an ESP-IDF loaded terminal.
+
+2. Run the `idf.py save-defconfig` command to generate a `sdkconfig.defaults` file.
+
+3. Use the `ESP-IDF: Open Project Configuration` and create a new profile with name `production`. Set `sdkconfig defaults` the previous `sdkconfig.defaults` file. If you want to separate generated binaries of this new **production** profile from the default case, specify a different build folder using the `Build Directory path` to something like `/path/to/current/project/build_production` and the `sdkconfig file path` to something like `/path/to/current/project/build_production/sdkconfig` to avoid mixing development sdkconfig with production sdkconfig files.
```

Comment:
- "sdkconfig file path" should be renamed "sdkconfig build path"

- shouldn't `sdkconfig defaults` be set to sdkconfig.defaults + sdkconfig.prod? 

- i'm confused why it is called `sdkconfig defaults`. Shouldn't it be called `sdkconfigPath`? It does not have to be the `sdkconfig.defaults` file, does it?


- "Use the 'ESP-IDF: Open Project
Configuration' and create a new profile with name 'production'. Set sokconfig defaults' the previous 'sdkconfig.defaults' file, for example XXXXXXXXX"

- "if you want to separate generated binaries..." should be rephrased, "Some people find it clearer to use a separate build folder for each project, to avoid confusion. for project 'prod' produces /build_prod/, and project YYYYYY for produces YYYYYY. This can be achieved by ...."


- ^^ also, why not just make the above the default? I think 99% of people want this setting